File: app/src/main/java/com/example/user/fuckinggreatadvicegetter/DBController.java
package com.example.user.fuckinggreatadvicegetter;

import io.realm.Realm;

class DBController {

    private Realm realm;

    public void add(String string) {
        Advice advice = new Advice();
        advice.setText(string);
        realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        realm.copyToRealm(advice);
        realm.commitTransaction();
        realm.close();
    }

    public void remove(String id) {
        realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        try {
            realm.where(Advice.class).equalTo("id", id).findFirst().deleteFromRealm();
        } catch (NullPointerException e) {
            e.printStackTrace();
        }
        realm.commitTransaction();
        realm.close();
    }
}

File: app/src/main/java/com/example/user/fuckinggreatadvicegetter/model/database/DBController.kt
package com.example.user.fuckinggreatadvicegetter.model.database

import com.example.user.fuckinggreatadvicegetter.model.Advice
import io.realm.Realm
import io.realm.RealmResults

class DBController {

    fun add(advice: Advice) {
        val realm = Realm.getDefaultInstance()
        realm.beginTransaction()
        realm.copyToRealm(advice)
        realm.commitTransaction()
        realm.close()
    }

    fun remove(id: String) {
        val realm = Realm.getDefaultInstance()
        realm.beginTransaction()
        realm.where(Advice::class.java).equalTo("id", id).findFirst()?.deleteFromRealm()
        realm.commitTransaction()
        realm.close()
    }

    fun findAll(): RealmResults<Advice>? {
        val realm = Realm.getDefaultInstance()
        return realm.where(Advice::class.java).findAll()
    }
}